llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5027,6 +5025,9 @@ calculateRegisterUsage(VPlan &Plan, ArrayRef<ElementCount> VFs,
         if (isa<VPVectorPointerRecipe, VPVectorEndPointerRecipe,
                 VPBranchOnMaskRecipe>(R))
           continue;
+        if (auto *Phi = dyn_cast<VPReductionPHIRecipe>(R);
```

Comment:
nit: could you add a short comment describing the reason for this?

Reply:
Won't in-loop reductions still carry a scalar register across iterations? Could we just reuse the scalar check below for scalar-use-only recipes:

```c++
            (isa<VPInstruction, VPReductionPHIRecipe>(R) &&
             all_of(cast<VPSingleDefRecipe>(R)->users(), [&](VPUser *U) {
               return cast<VPRecipeBase>(U)->usesScalars(R->getVPSingleValue());
             }))) {
```

Reply:
That's a much better way of dealing with it, thank you. Done.